src/resolve/resolved-varlink.c

```diff
@@ -519,6 +535,31 @@ static int vl_method_resolve_address(Varlink *link, JsonVariant *parameters, Var
         return 1;
 }
 
+static int vl_method_subscribe_dns_resolves(Varlink *link, JsonVariant *parameters, VarlinkMethodFlags flags, void *userdata)
+{
+        Manager *m;
+        int r;
+
+        assert(link);
+
+        m = varlink_server_get_userdata(varlink_get_server(link));
+        assert(m);
+
+        if (json_variant_elements(parameters) > 0)
+                return varlink_error_invalid_parameter(link, parameters);
+
+        r = set_ensure_put(&m->varlink_subscription, NULL, varlink_ref(link));
+        if (r < 0)
+                return log_error_errno(r, "Failed to add subscription to set: %m");
+
+        /* if the client didn't set the more flag, return an empty response and close the connection */
+        if (!FLAGS_SET(flags, VARLINK_METHOD_MORE))
+                return varlink_reply(link, NULL);
+
+        log_debug("%u clients now attached for varlink notifications", set_size(m->varlink_subscription));
```

Comment:
Maybe, we should add the log before `varlink_reply()` in the above?

Reply:
I've moved this in the latest patch now